### PR TITLE
memory_growth_to_js_library

### PR DESCRIPTION
**Pull request by juj**
Move reallocBuffer and emscripten_replace_memory to be handled in JS library functions

### CODE REVIEW

File: src/library.js [hunk 1]
```diff
@@ -479,7 +479,57 @@ LibraryManager.library = {
...
   },
 
-  emscripten_resize_heap__deps: ['emscripten_get_heap_size', '$abortOnCannotGrowMemory'],
+#if TEST_MEMORY_GROWTH_FAILS
+  _reallocBuffer: function(size) {
+    return null;
+  },
+#else
+
+  _reallocBuffer: function(size) {
```
Comment (kripken) [inline src/library.js]: how about calling this `$emscripten_realloc_buffer`?

File: tests/optimizer/eliminateDeadGlobals-output.js [hunk 2]
```diff
@@ -17,19 +17,6 @@ Module["asm"] = (function(global, env, buffer) {
  var _memcpy = env._memcpy;
  var __ZdaPv = env.__ZdaPv;
  var __ZdlPv = env.__ZdlPv;
- function _emscripten_replace_memory(newBuffer) {
```
Comment (kripken) [inline tests/optimizer/eliminateDeadGlobals-output.js]: I don't think we need to remove this from the tests, unless you think it's potentially confusing?

File: tests/test_other.py [hunk 3]
```diff
@@ -7851,9 +7839,9 @@ def test(filename, expectations, size_slack):
 
       print('test on hello world')
       test(path_from_root('tests', 'hello_world.cpp'), [
-        ([],      21, ['assert'], ['waka'], 46505,  22,   15, 58), # noqa
```
Comment (kripken) [inline tests/test_other.py]: why did this increase? I thought this just moved code around, but left the same imports and exports from the asm module?

File: src/library.js (hunk 1 above)
Comment (juj) [reply src/library.js]: Ok.

File: tests/test_other.py (hunk 3 above)
Comment (juj) [reply tests/test_other.py]: Oh, sloppy __deps handling, had that included also in ALLOW_MEMORY_GROWTH=0 builds, fixed now.

File: tests/optimizer/eliminateDeadGlobals-output.js (hunk 2 above)
Comment (juj) [reply tests/optimizer/eliminateDeadGlobals-output.js]: Check, restored back. Originally I just search-replace removed all occurrences, but certainly doesn't matter to leave this here.

**Pull request merged**